rust_src/remacs-sys/lib.rs

```diff
@@ -1040,6 +1042,8 @@ extern "C" {
     ) -> ptrdiff_t;
 
     pub fn hash_remove_from_table(h: *mut Lisp_Hash_Table, key: Lisp_Object);
+
+    pub fn bset_update_mode_line(buffer: *const libc::c_void) -> Lisp_Object; // TODO: buffer*
```

Comment:
You can use `*mut Lisp_Buffer`.

Reply:
would it make sense to change this for set_buffer_internal as well?

Reply:
Yes, I guesss that function predates our definition of the struct.